docs/deep-dive/substances.md

```diff
@@ -0,0 +1,88 @@
+# Substances on a high level
+
+The main domain object in a LIMS is probably the Sample object or objects derived from samples (aliquots). In Common LIMS, samples are subclasses of a more generic type called Substance. A `Substance` models both Samples and Aliquots, but also other material that may be combined with them in some lab-specific set.
+
+Let's imagine that you have a gemstone lab. In this lab, a sample might be a small bit of a larger gemstone.
+
+You would have to tell the LIMS that you're working with gemstones, so you would implement the GemstoneSample type in your plugin:
+
+```
+# TODO: This part of the API is still being worked on. One can currently only
+# interact with the proxied objects mentioned below
+
+class GemstoneSample(Sample):
+    color = TextField()
+    comment = TextField()
+```
+
+The GemstoneSample object is in fact a proxy for a range of database models:
+
+    * ExtensibleType
+    * ExtensiblePropertyType
+    * Substance
+    * ExtensibleProperty
+
```

Comment:
In code, there is ExtensiblePropertyValue as well. I think it would be nice with a description of how ExtensiblePropertyValue and ExtensibleProperty relates to each other, and also how version of a property comes in. I think it would be really helpful to have this explanation to lean back on when one want to refresh the knowledge of how it works.

Reply:
Good point! I'll fix this.

Reply:
I've added `./deep-dive/extensible.md` that goes into detail. Check out my rebased commit.